Sources/Apodini/Request/Parameter.swift

```diff
@@ -0,0 +1,138 @@
+    ///     @Parameter var birdID: Bird.ID
+    ///     // Exposed as a lightweight parameter
+    ///     @Parameter var nestName: String?
+    ///
+    ///     // ...
+    /// }
+    ///
+    /// struct TestWebService: WebService {
+    ///     @Parameter var birdID: Bird.ID
+    ///
+    ///     var content: some Component {
+    ///         Group("api", "birds", birdID) {
+    ///             ExampleBirdComponent(birdID: $birdID)
+    ///             Group("nests") {
+    ///                 ExampleNestComponent(birdID: $birdID)
+    ///             }
+    ///         }
+    ///     }
+    /// }
+    ///
+    /// TestWebService.main()
+    /// ```
+    public var projectedValue: Parameter {
+        self
+    }
+    
+    
+    /// Creates a new `@Parameter` that indicates input of a `Component` without a default value, different name for the encoding, or special options.
+    public init() {
+        self.defaultValue = nil
+        self.name = nil
+        self.options = PropertyOptionSet()
+    }
+    
+    /// Creates a new `@Parameter` that indicates input of a `Component`.
+    /// - Parameters:
+    ///   - name: The name that identifies this property when decoding the property from the input of a `Component`
+    ///   - options: Options passed on to different interface exporters to clarify the functionality of this `@Parameter` for different API types
+    public init(_ name: String? = nil, _ options: Option...) {
+        self.defaultValue = nil
+        self.name = name
+        self.options = PropertyOptionSet(options)
+    }
+    
+    /// Creates a new `@Parameter` that indicates input of a `Component`.
+    /// - Parameters:
+    ///   - defaultValue: The default value that should be used in case the interface exporter can not decode the value from the input of the `Component`
+    ///   - name: The name that identifies this property when decoding the property from the input of a `Component`
+    ///   - options: Options passed on to different interface exporters to clarify the functionality of this `@Parameter` for different API types
+    public init(wrappedValue defaultValue: Element, _ name: String? = nil, _ options: Option...) {
+        self.defaultValue = defaultValue
+        self.name = name
+        self.options = PropertyOptionSet(options)
+    }
+    
+    /// Creates a new `@Parameter` that indicates input of a `Component` based on an existing component.
+    /// - Parameter pathParameter: The `@PathParameter` that can be passed in from a a parent component.
+    /// - Precondition: A `@Parameter` with a specific `http` type `.body` or `.query` can not be passed to a seperate componet. Please remove the specific `.http` property option or specify the `.http` property option to `.path`.
+    public init(_ pathParameter: PathParameter<Element>) {
+        self.options = PropertyOptionSet([.http(.path)])
+        self.id = pathParameter.id
+    }
```

Comment:
Can we make the initializer private/internal? I don't think this should ever be invoked by the user right?

Reply:
I will check if we can make this private or internal. This initializer is used when passing down a `@PathParameter` down the component tree using the projected value. I will get back to you on this 👍

Reply:
I have updated the initializer in the latest commit, it is now internal.